### ISSUE TITLE
mon fails due to systemd unit file error

### ISSUE THREAD

**Issue by alitvak69** [ceph/ceph-ansible#3064]
**Bug Report**

What happened:
systemd failed to start 1st ceph mon
What you expected to happen:
systemd successfully started 1st ceph mon
How to reproduce it (minimal and precise):

Run deployment and observe it fail at this task
2018-08-26 09:15:33,260 p=604 u=vagrant |  TASK [ceph-mon : systemd start mon container] ***************************************************************************************
2018-08-26 09:15:33,260 p=604 u=vagrant |  task path: /home/vagrant/ansible/alex/ceph-ansible/roles/ceph-mon/tasks/docker/start_docker_monitor.yml:54
2018-08-26 09:15:33,261 p=604 u=vagrant |  Sunday 26 August 2018  09:15:33 +0000 (0:00:01.559)       0:04:34.476 ********* 
2018-08-26 09:15:33,304 p=604 u=vagrant |  Using module file /usr/lib/python2.7/dist-packages/ansible/modules/system/systemd.py
2018-08-26 09:15:33,627 p=604 u=vagrant |  fatal: [storage1n1-chi]: FAILED! => {
    "changed": false, 
    "invocation": {
        "module_args": {
            "daemon_reload": true, 
            "enabled": true, 
            "force": null, 
            "masked": null, 
            "name": "ceph-mon@storage1n1-chi", 
            "no_block": false, 
            "state": "started", 
            "user": false
        }
    }, 
    "msg": "Error loading unit file 'ceph-mon@storage1n1-chi': System.Error.ENOEXEC \"Exec format error\""

system logs show:

Aug 26 09:33:43 storage1n1-chi systemd[1]: **/etc/systemd/system/ceph-mon@.service:8: Executable path is not absolute: $(command -v mkdir) -p /etc/ceph /var/lib/ceph/mon**
Aug 26 09:33:43 storage1n1-chi systemd[1]: **ceph-mon@storage1n1-chi.service: Unit configuration has fatal error, unit will not be started**

I fixed by changing a line in template ceph-mon.service.j2

 [Service]
EnvironmentFile=-/etc/environment 
ExecStartPre=-/usr/bin/docker rm ceph-mon-%i 
--  ExecStartPre=$(command -v mkdir) -p /etc/ceph /var/lib/ceph/mon 
++ ExecStartPre=/bin/bash -c '$$(command -v mkdir) -p /etc/ceph /var/lib/ceph/mon' 
ExecStart=/usr/bin/docker run --rm --name ceph-mon-%i \

Share your group_vars files, inventory

**Environment**:
* OS (e.g. from /etc/os-release):
NAME="Ubuntu"
VERSION="18.04.1 LTS (Bionic Beaver)"

* Kernel (e.g. `uname -a`):
4.15.0-32-generic
* Docker version if applicable (e.g. `docker version`):
Client:
 Version:	17.12.1-ce
 API version:	1.35
 Go version:	go1.10.1
 Git commit:	7390fc6
 Built:	Wed Apr 18 01:23:11 2018
 OS/Arch:	linux/amd64

Server:
 Engine:
  Version:	17.12.1-ce
  API version:	1.35 (minimum version 1.12)
  Go version:	go1.10.1
  Git commit:	7390fc6
  Built:	Wed Feb 28 17:46:05 2018
  OS/Arch:	linux/amd64
  Experimental:	false

* Ansible version (e.g. `ansible-playbook --version`):
ansible-playbook 2.6.3
* ceph-ansible version (e.g. `git head or tag or stable branch`):
master
* Ceph version (e.g. `ceph -v`):
13,2,1 (mimic)

**Comment by leseb** [comment]
I wonder if the `command -v mkdir` is even necessary... Let's test without it.
Thanks for reporting the issue.

**Comment by guits** [comment]
@alitvak69 that's weird, I've tried your syntax, it ends up with :
```
Sep 06 14:09:58 mon0.ceph.redhat.com bash[668]: /bin/bash: -c: option requires an argument
```

The only syntax i've been able to get working is this one (using '$$' or '$' has the same result in this one):

```
ExecStartPre=/bin/sh -c '"$(command -v mkdir)" -p /etc/ceph /var/lib/ceph/mon'
```

@alitvak69 do you mind testing this one in your env?

Thanks!

**Comment by alitvak69** [comment]
It is odd because while installation worked for me multiple times, shell command with $$ returns an error in bionic.

/bin/sh -c '$$(command -v mkdir) -p /etc/ceph /var/lib/ceph/mon'
/bin/sh: 1: Syntax error: word unexpected (expecting ")")

/bin/bash -c '$$(command -v mkdir) -p /etc/ceph /var/lib/ceph/mon'
/bin/bash: -c: line 0: syntax error near unexpected token `command'
/bin/bash: -c: line 0: `$$(command -v mkdir) -p /etc/ceph /var/lib/ceph/mon'

However, both /bin/sh and /bin/bash work fine with a single $

**Comment by guits** [comment]
Closed by #3101

**Comment by alitvak69** [comment]
Thinking about it, I used someone else suggestion in the similar situation.
https://stackoverflow.com/questions/48684468/executable-path-is-not-absolute-ignoring-which-node

However, $$(xxx xx) sounds like a wrong bash syntax and it should be $(xxx xxx)
